manifests/vrrp/track_process.pp

```diff
@@ -20,12 +20,17 @@
 # $full_command::  Match entire process cmdline
 #             Default: undef
 #
+# $param_match::  Set inital if command have no parameters or
+#             use partial if first n parameters match
+#             Default: undef
+#
 define keepalived::vrrp::track_process (
   String[1] $proc_name,
-  Optional[Integer[0]] $weight   = undef,
-  Integer[0] $quorum             = 1,
-  Optional[Integer[0]] $delay    = undef,
-  Boolean $full_command           = false
+  Optional[Integer[0]] $weight           = undef,
+  Integer[0] $quorum                     = 1,
+  Optional[Integer[0]] $delay            = undef,
+  Boolean $full_command                  = false,
+  Enum['initial','partial'] $param_match = undef
```

Comment:
Catalog Compilation will fail with default params, because an Enum can't be undef. Please use `Optional[Enum['initial','partial']]`

Reply:
Added type Optional.